sonar-common/src/main/java/xyz/jonesdev/sonar/common/fallback/session/FallbackGravitySessionHandler.java

```diff
@@ -0,0 +1,261 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package xyz.jonesdev.sonar.common.fallback.session;
+
+import lombok.val;
+import org.jetbrains.annotations.NotNull;
+import xyz.jonesdev.sonar.api.Sonar;
+import xyz.jonesdev.sonar.api.fallback.FallbackUser;
+import xyz.jonesdev.sonar.common.fallback.protocol.FallbackPacket;
+import xyz.jonesdev.sonar.common.fallback.protocol.FallbackPacketDecoder;
+import xyz.jonesdev.sonar.common.fallback.protocol.packets.play.*;
+
+import java.util.UUID;
+
+import static xyz.jonesdev.sonar.api.config.SonarConfiguration.Verification.Gravity.Gamemode.CREATIVE;
+import static xyz.jonesdev.sonar.api.fallback.protocol.ProtocolVersion.*;
+import static xyz.jonesdev.sonar.common.fallback.protocol.FallbackPreparer.*;
+
+public final class FallbackGravitySessionHandler extends FallbackSessionHandler {
+
+  public FallbackGravitySessionHandler(final FallbackUser user, final String username, final UUID uuid) {
+    super(user, username, uuid);
+
+    // We don't want to check Geyser players for valid gravity, as this might cause issues because of the protocol
+    this.enableGravityCheck = !user.isGeyser() && Sonar.get().getConfig().getVerification().getGravity().isEnabled();
+    this.enableCollisionsCheck = Sonar.get().getConfig().getVerification().getGravity().isCheckCollisions();
+
+    // FIXME: 1.18.2-1.19.3 weird loading world issue
+    // First, write the JoinGame packet to the buffer
+    user.delayedWrite(joinGame);
+    // Then, write the ClientAbilities packet to the buffer
+    // This is only necessary if the player is in creative mode
+    if (Sonar.get().getConfig().getVerification().getGravity().getGamemode() == CREATIVE) {
+      user.delayedWrite(DEFAULT_ABILITIES);
+    }
+    // Write the PositionLook packet to the buffer
+    user.delayedWrite(spawnPosition);
+    // Write the DefaultSpawnPosition packet to the buffer
+    if (user.getProtocolVersion().compareTo(MINECRAFT_1_19_3) >= 0) {
+      user.delayedWrite(defaultSpawnPosition);
+    }
+    // Write the PlayerInfo packet to the buffer
+    if (user.getProtocolVersion().compareTo(MINECRAFT_1_16_4) >= 0) {
+      user.delayedWrite(PLAYER_INFO);
+    }
+    // 1.20.3+ introduced game events
+    // Make sure the client knows that we're sending chunks next
+    if (user.getProtocolVersion().compareTo(MINECRAFT_1_20_3) >= 0) {
+      user.delayedWrite(START_WRITING_CHUNKS);
+    }
+    // Teleport player into an empty world by sending an empty chunk packet
+    user.delayedWrite(EMPTY_CHUNK_DATA);
+    // Spawn the invisible platform below the player
+    if (this.enableCollisionsCheck) {
+      user.delayedWrite(updateSectionBlocks);
+    }
```

Comment:
Related to the 1.18.2-1.19.3 bug

Reply:
I'll take a look at this later when I have some time for it. 1.18.2-1.19.3 can verify without problems, this is just a *potential* issue.